CTFd/events/__init__.py

```diff
@@ -13,6 +15,19 @@ def subscribe():
     @stream_with_context
     def gen():
         for event in current_app.events_manager.subscribe():
+            # Filter notifications based on user and team ID
+            if event.type == "notification":
+                if (
+                    event.data["user"] is not None
+                    and event.data["user"] != get_current_user().id
+                ):
+                    continue
+                if get_config("user_mode") == TEAMS_MODE:
+                    if (
+                        event.data["team"] is not None
+                        and event.data["team"] != get_current_team().id
+                    ):
+                        continue
```

Comment:
This code should be moved into the event managers and allow for subscribing to events that have the specified user/team IDs. You should use `data["user_id"]` instead of `["user"]`. Same with team.